skyvern/forge/agent.py

```diff
@@ -1773,6 +1774,24 @@ async def update_step(
             step_id=step.step_id,
             diff=update_comparison,
         )
+
+        try:
+            log = skyvern_context.current().log
+            current_step_log = [entry for entry in log if entry.get("step_id", "") == step.step_id]
+            log_json = json.dumps(current_step_log, cls=SkyvernLogEncoder, indent=2)
+            await app.ARTIFACT_MANAGER.create_artifact(
```

Comment:
Is there a way we can also capture task and workflow level logs? Even if we don't render them in the UI today

We historically associate logs like that w/ the first step, although @wintonzheng is currently overhauling it

Reply:
yeah we just added new columns to the artifacts db: https://github.com/Skyvern-AI/skyvern/blob/main/skyvern/forge/sdk/db/models.py#L169-L174

Reply:
https://github.com/Skyvern-AI/skyvern/pull/1345/files

here's the example of supporting artifact upload for observer thought.

going forward we can add support for task level and workflow run level artifacts